cumulus_library/studies/core/count_core.sql

```diff
@@ -0,0 +1,151 @@
+-- ###########################################################
+CREATE or replace VIEW core__count_patient AS 
+    with powerset as
+    (
+        select
+        count(distinct subject_ref)   as cnt_subject
+        
+        , age, gender, race_display, ethnicity_display        
+        FROM core__patient
+        group by CUBE
+        ( age, gender, race_display, ethnicity_display )
+    )
+    select
+          cnt_subject as cnt 
+        , age, gender, race_display, ethnicity_display
+    from powerset 
+    WHERE cnt_subject >= 10 
+    ORDER BY cnt desc;
+
+-- ###########################################################
+CREATE or replace VIEW core__count_procedure AS 
+    with powerset as
+    (
+        select
+        count(distinct subject_ref)   as cnt_subject
+        , count(distinct encounter_ref)   as cnt_encounter
+        , proc_display, proc_system        
+        FROM core__procedure
+        group by CUBE
+        ( proc_display, proc_system )
+    )
+    select
+          cnt_encounter  as cnt 
+        , proc_display, proc_system
+    from powerset 
+    WHERE cnt_subject >= 10 
+    ORDER BY cnt desc;
+
+-- ###########################################################
+CREATE or replace VIEW core__count_encounter_demographics_month AS 
+    with powerset as
+    (
+        select
+        count(distinct subject_ref)   as cnt_subject
+        , count(distinct encounter_ref)   as cnt_encounter
+        , start_month, enc_class_display, age_at_visit, gender, race_display, ethnicity_display        
+        FROM core__encounter
+        group by CUBE
+        ( start_month, enc_class_display, age_at_visit, gender, race_display, ethnicity_display )
+    )
+    select
+          cnt_encounter  as cnt 
+        , start_month, enc_class_display, age_at_visit, gender, race_display, ethnicity_display
+    from powerset 
+    WHERE cnt_subject >= 10 
+    ORDER BY cnt desc;
+
+-- ###########################################################
+CREATE or replace VIEW core__count_encounter_type AS 
```

Comment:
checking - are these encounter counts intentionally using a different binning minimum?

Reply:
yes, because encounters are for everyone and we dont want to overwhelm with small bin sizes. This also allows us to run "monthly" counts which is impossible (athena query dies) when we set the min size to 10